tests/kernel/smp/src/main.c

```diff
@@ -632,59 +630,57 @@ void test_smp_ipi(void)
 	}
 }
 
-void ztest_post_fatal_error_hook(unsigned int reason, const z_arch_esf_t *pEsf)
+void k_sys_fatal_error_handler(unsigned int reason, const z_arch_esf_t *pEsf)
 {
-	static int times;
+	static int trigger;
 
 	if (reason != K_ERR_KERNEL_OOPS) {
 		printk("wrong error reason\n");
 		k_fatal_halt(reason);
 	}
 
-	if (times == 0) {
-		main_thread_id = curr_cpu();
-		times++;
-	} else {
+	if (trigger == 0) {
 		child_thread_id = curr_cpu();
+		trigger++;
+	} else {
+		main_thread_id = curr_cpu();
+
+		/* Verify the fatal was happened on different core */
+		zassert_true(main_thread_id != child_thread_id,
+					"fatal on the same core");
 	}
 }
 
 void entry_oops(void *p1, void *p2, void *p3)
 {
-	unsigned int key;
-
-	ztest_set_fault_valid(true);
-
-	key = irq_lock();
 	k_oops();
 	TC_ERROR("SHOULD NEVER SEE THIS\n");
-	rv = TC_FAIL;
-	irq_unlock(key);
```

Comment:
Seems like this will also resolve #35202

Indeed, I don't understand why that was every trying to oops with a lock held.

Reply:
Yes, @andyross , I think so, too. And thanks so much for your clear descriptions in #35202 . :)